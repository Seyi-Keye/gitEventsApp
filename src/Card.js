import React, { Component } from "react";
import Avatar from "./Avatar";
import Information from "./Information";

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      error: "",
      username: this.props.username
    };
  }

  componentDidMount() {
    const { username } = this.props;
    this.fetchEventPayload(username);
  }

  componentWillReceiveProps(nextProps) {
    this.fetchEventPayload(nextProps.username);
  }

  handleDisplay = data => {
    const result = data.map((datum, index) => {
      return (
        <div
          key={index.toString()}
          style={{
            display: "flex",
            border: "5px solid white",
            background: "grey",
            width: "70%",
            // boxSizing: "border-box",
            margin: "auto",
            padding: "10px",
            height: "70px"
          }}
        >
          <Avatar picture={datum.actor} />
          <Information data={datum} />
        </div>
      );
    });
    return result;
  };

  fetchEventPayload = username => {
    fetch(`https://api.github.com/users/${username}/events`)
      .then(data => {
        if (data.status !== 200) {
          throw new Error(data.statusText);
        }
        return data.json();
      })
      .then(data => {
        this.setState({ data });
      })
      .catch(error => this.setState({ error }));
  };

  checkError = error => {
    return (
      <p style={{ padding: "10px", color: "red", fontSize: "20px" }}>
        {error.message}
      </p>
    );
  };

  render() {
    const data = this.state.data;
    const error = this.state.error;
    return (
      <div>
        {data && this.handleDisplay(data)}
        {error && this.checkError(error)}
        {!error && data.length === 0 && <p> No results </p>}
      </div>
    );
  }
}
export default Card;
