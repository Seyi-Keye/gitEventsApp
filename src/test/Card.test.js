import React from "react";
import Card from "../Card";
import renderer from "react-test-renderer";

const initialState = {
  data: [],
  error: ""
};

describe("Card component", () => {
  let wrapper = renderer.create(<Card />);
  it("renders succesfully", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders without props", () => {
    const testInstace = wrapper.root;
    expect(testInstace.props).toEqual({});
  });

  it("renders correct initial state", () => {
    const testInstance = wrapper.root;
    expect(testInstance.instance.state).toEqual(initialState);
  });
});

describe("Card component", () => {
  let wrapper = renderer.create(<Card />);
  const testInstance = wrapper.root.instance;
  testInstance.setState({ username: "aromokeye" });

  it("renders succesfully with props", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders with props", () => {
    expect(testInstance.state).toEqual({
      data: [],
      error: "",
      username: "aromokeye"
    });
  });
});

// fdescribe("Card functionalities", () => {
//   let fetchEventPayload = false;
//   // const node = {
//   //   createNodeMock: () => {
//   //     return {
//   //       handleFormSubmit: () => {
//   //         focused = true;
//   //       }
//   //     };
//   //   }
//   // };

//   const stub = () => true;

//   const componentRoot = renderer.create(<Card />).root;
//   // , node).root;

//   const componentInstance = componentRoot.instance;
//   componentInstance.setState({ username: "aromokeye" });
//   const fetchEventPayloadSpy = jest.spyOn(
//     componentInstance.prototype,
//     "fetchEventPayload"
//   );

//   console.log(
//     "button",
//     fetchEventPayloadSpy,
//     "========",
//     componentInstance.fetchEventPayload
//   );
//   it("calls fetchEventPayload", () => {
//     expect(fetchEventPayloadSpy).toHaveBeenCalled();
//   });
// });
