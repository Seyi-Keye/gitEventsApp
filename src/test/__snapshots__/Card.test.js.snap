exports[`Card component renders succesfully 1`] = `
<div>
  
  <p>
     No results 
  </p>
</div>
`;

exports[`Card component renders succesfully with props 1`] = `
<div>
  
  <p>
     No results 
  </p>
</div>
`;
